Add CI with selftests in a container

Werkov opened the pull request:
Followup to [this](https://github.com/rapido-linux/rapido/issues/254#issuecomment-3175282786)

One could install `kernel-default-devel` and `KERNEL_SRC=/usr/src/linux-6.16.3-1-obj/x86_64/default/` + tweak lookup of vmlinuz. Or teach rapido to use specific kernel version (commit in this PR) and then run the tests in a simple container image:

```Dockerfile
FROM registry.opensuse.org/opensuse/tumbleweed:latest
RUN zypper --non-interactive in dracut expect kernel-default 
RUN zypper --non-interactive in qemu-x86

# only needed with KERNEL_SRC override, won't work without further changes in rapido
# RUN zypper --non-interactive in kernel-source
# RUN zypper --non-interactive in kernel-devel
# RUN zypper --non-interactive in kernel-syms

CMD /test/selftest/selftest.sh
```

And then I can run selftests like:
```
docker run -v $PWD:/test -w /test \
    -e KERNEL_RELEASE=6.16.3-1-default
    -e KERNEL_INSTALL_MOD_PATH=/usr/
    localhost/rapido-test-image
```

### Next up:

- push the image somewhere
- [figure out how to make Github CI](https://stackoverflow.com/a/64200200) run actions inside said image
- possibly automate building/refreshing the image + automatic updated of KERNEL_RELEASE used in CI

Werkov committed:
runtime: Allow overriding kernel version

When we run inside a container, we cannot infer much from `uname -r`. We
can install abitrary kernel inside the image, the easiest way to work
with that kernel is to tell rapido exactly what kernel release it should
look for.

Signed-off-by: Michal Koutný <mkoutny@suse.com>

ddiss commented:
The new setting looks reasonable to me. Please document its use in `rapido.conf.example`. The same setting should be usable for "regular" Linux hosts with multiple kernel versions installed, right?

thackara commented:
if we had installed `kernel-default-devel`, the `KERNEL_RELEASE` can be read from filesystem rather than an ENV?
```
# cat /usr/src/linux-obj/x86_64/default/include/config/kernel.release
6.16.7-1-default
```

Werkov committed:
CI: Initial commit

Signed-off-by: Michal Koutný <mkoutny@suse.com>

Werkov commented:
I made it work in my fork of the repo (where I commited `.github/workflows/` into `master`. [See](https://github.com/Werkov/rapido/actions/runs/17864573572/job/50803653761?pr=1#step:4:506).

The test image is built and hosted in [OBS](https://build.opensuse.org/package/show/home:mkoutny:containers/rapido-ci). Now, I installed all with 'latest' version but then I have to put absolute number in `ci.yml` `KERNEL_RELEASE`. That would work if the container image was never rebuilt.

I may look into the suggestion with `kernel-default-devel` and filling it automatically inside this test container from `ci.yml`. (Although, I'm afraid it would not generalize for generic rapido use with any distro as `-devel` is openSUSE specific.)

Unmarking draft for more feedback.

ddiss commented on rapido.conf.example:
```diff
@@ -1,11 +1,16 @@
-# If KERNEL_SRC is not set then rapido will attempt to boot the local kernel
-# present under /boot/.
+# If KERNEL_SRC is not set then rapido will attempt to boot the local running
+# kernel present under /boot/ unless KERNEL_RELEASE is set.
```
I think I'd prefer:
```
# If KERNEL_SRC is not set then rapido will attempt to boot a host kernel
# available under /boot/. Either the `uname` release, or the version explicitly specified via KERNEL_RELEASE
```

Werkov committed:
CI: Detect version of installed kernel for selftests

Signed-off-by: Michal Koutný <mkoutny@suse.com>

Werkov commented:
Added `kernel-default-devel` into the container image for autodetection of `KERNEL_RELEASE `in CI. Addressed feedback on comments. CI restared in a [user fork](https://github.com/Werkov/rapido/pull/1).

ddiss commented on runtime.vars:
```diff
@@ -308,6 +308,8 @@ _rt_require_dracut_args() {
 		[[ -n $kver ]] \
 			|| _fail "failed to read kernel.release at $KERNEL_SRC"
+	elif [[ -n $KERNEL_RELEASE ]] ; then
+		kver=$KERNEL_RELEASE
```
The assignment here should be quoted. I'll add that when I merge.

ddiss reviewed (approved):
Looks good, thanks! Will merge with the minor quoting change squashed in.
